QmlNet/QmlNet/qml/NetQModelIndex.cpp

```diff
@@ -0,0 +1,22 @@
+#include <QModelIndex>
+
+#include "NetQModelIndex.h"
+
+extern "C" {
+
+Q_DECL_EXPORT int net_qmodelindex_column(NetQModelIndexContainer* container) {
+    return container->modelIndex->column();
+}
+Q_DECL_EXPORT int net_qmodelindex_row(NetQModelIndexContainer* container) {
+    return container->modelIndex->row();
+}
+Q_DECL_EXPORT NetQModelIndexContainer* net_qmodelindex_parent(NetQModelIndexContainer* container) {
+    return new NetQModelIndexContainer{
+        .modelIndex = QSharedPointer<QModelIndex>(&container->modelIndex->parent())
```

Comment:
How does the ownership work here?

When .NET references the parent, when it GCs the wrapper, it would dispose of the const reference incorrectly. No?